lib/order.rb

```diff
@@ -0,0 +1,86 @@
+require 'csv'
+
+class Order
+
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+
+    valid_fulfillments = [:pending, :paid, :processing, :shipped, :complete]
+    raise ArgumentError, "invalid fulfillment status" unless valid_fulfillments.include?(fulfillment_status)
```

Comment:
It's helpful to include the invalid argument in the messages of your `ArgumentError`s:

```suggestion
    raise ArgumentError, "invalid fulfillment status: #{fulfillment_status}" unless valid_fulfillments.include?(fulfillment_status)
```